app/repositories/claim-repository.js

```diff
@@ -388,13 +389,36 @@ export const addHerdToClaimData = async (reference, herdId, herdVersion, herdAss
   const [_, _updates] = await buildData.models.claim.update(
     {
       data,
-      updatedBy: user
+      updatedBy: createdBy
     },
     {
-      where: { reference },
+      where: { reference: claimRef },
       returning: true
     }
   )
+
+  await raiseHerdEvent({
```

Comment:
I think this call and the one below should be swapped round. It's extremely unlikely obviously but here we might emit event the herd was associated but then fail to update the DB so it's not actually been associated